### PR TITLE
8264166: OopStorage should support specifying MEMFLAGS for allocations

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this change to OopStorage to allow the MEMFLAGS value for associated allocations to be specified when the storage object is constructed.  This allows a subsystem that needs an OopStorage object to associate its allocation with others for that subsystem in NMT tracking and reporting.

Testing:
mach5 tier1.
Manually compared NMT output before and after this change for a test that generated a lot of one particular OopStorage entries.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264166](https://bugs.openjdk.java.net/browse/JDK-8264166): OopStorage should support specifying MEMFLAGS for allocations

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to a9c808be9f0a417109116985e1ca92cc54b0602a
 * [Stefan Karlsson](https://openjdk.java.net/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to a9c808be9f0a417109116985e1ca92cc54b0602a

### Download
To checkout this PR locally:
`$ git fetch https://git.openjdk.java.net/jdk pull/3188/head:pull/3188`
`$ git checkout pull/3188`

To update a local copy of the PR:
`$ git checkout pull/3188`
`$ git pull https://git.openjdk.java.net/jdk pull/3188/head`

### CODE REVIEW

**Review by tschatzl — APPROVED** [review]
Lgtm.

File: src/hotspot/share/gc/shared/oopStorage.inline.hpp [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #ifndef SHARE_GC_SHARED_OOPSTORAGE_INLINE_HPP
 #define SHARE_GC_SHARED_OOPSTORAGE_INLINE_HPP
 
+#include "memory/allocation.hpp"
```
Comment (stefank) [inline src/hotspot/share/gc/shared/oopStorage.inline.hpp:28]: Sort order

**Review by stefank — APPROVED** [review]
Looks good. Just one nit

File: src/hotspot/share/gc/shared/oopStorage.inline.hpp (hunk 1 above)
Comment (kimbarrett) [reply src/hotspot/share/gc/shared/oopStorage.inline.hpp:28]: Oops, will fix.

**Pull request closed without merging**